cypress/integration/data-integrity/index.js

```diff
@@ -0,0 +1,47 @@
+import { Before, Given, When, Then } from 'cypress-cucumber-preprocessor/steps'
+
+const JOB_ID = 'JOB_ID'
+
+Before(() => {
+    cy.clock()
+
+    cy.intercept({ pathname: /dataIntegrity$/, method: 'POST' }, req => {
+        req.reply(200, {
+            response: {
+                id: JOB_ID,
+            },
+        })
+    })
+
+    cy.intercept(
+        { pathname: /system\/taskSummaries\/DATA_INTEGRITY/, method: 'GET' },
+        req => {
+            expect(req.url.endsWith(JOB_ID)).to.be.true
+            req.reply(200, {
+                errors: ['Element 1', 'Element 2', 'Element 3'],
+                errorsWithSubheaders: {
+                    subheader: ['Value 1', 'Value 2'],
+                },
+                passed: {},
+            })
+        }
+    ).as('pollDataIntegrityCheck')
+})
+
+Given('the user navigated to the data integrity page', () => {
+    cy.visit('/#/data-integrity')
+    cy.findByRole('heading', { name: 'Data Integrity' }).should('exist')
+})
+
+When(`the user clicks on the 'Run integrity checks' button`, () => {
+    cy.findByRole('button', { name: 'Run integrity checks' }).click()
+})
+
+Then('a data integrity check is requested', () => {
+    cy.tick(5000)
+    cy.wait('@pollDataIntegrityCheck')
+})
+
+Then('the user is shown the results of the check', () => {
+    // TODO
```

Comment:
Is this a todo we should resolve in the current PR? Or after that?

Reply:
Both - I was unsure of the best way to test this and I was hesistant to try as the issue cards are material UI components which we should replace with `@dhis2/ui` components (which would change the way we test this). For now I've added some basic tests that only verify the card titles, but once we migrate away from material UI it'd be nice to also check the content of the cards (shown when they are expanded by the user).